test/helpers/helpers.go

```diff
@@ -91,27 +91,30 @@ func NewMachineConfigExtended(
 	if annotations == nil {
 		annotations = map[string]string{}
 	}
-	rawIgnition := MarshalOrDie(
-		&ign3types.Config{
-			Ignition: ign3types.Ignition{
-				Version: ign3types.MaxVersion.String(),
-			},
-			Storage: ign3types.Storage{
-				Files: files,
-			},
-			Systemd: ign3types.Systemd{
-				Units: units,
-			},
-			Passwd: ign3types.Passwd{
-				Users: []ign3types.PasswdUser{
-					{
-						Name:              "core",
-						SSHAuthorizedKeys: sshkeys,
-					},
+
+	ignConfig := &ign3types.Config{
+		Ignition: ign3types.Ignition{
+			Version: ign3types.MaxVersion.String(),
+		},
+		Storage: ign3types.Storage{
+			Files: files,
+		},
+		Systemd: ign3types.Systemd{
+			Units: units,
+		},
+		Passwd: ign3types.Passwd{},
+	}
+
+	if len(sshkeys) != 0 {
+		ignConfig.Passwd = ign3types.Passwd{
```

Comment:
Note to reviewer: I tested this change in isolation here https://github.com/openshift/machine-config-operator/pull/4172 to validate that it would not have any unforeseen side effects.